### PR TITLE
Set new compilation flags from version 4.2.0

### LINKED ISSUE #611: Adapt SPECS to the compilation changes from Wazuh core
Hi team,

This issue describes some changes that will be applied at the Wazuh's compilation that may affect the package building process. High-level changes are described at https://github.com/wazuh/wazuh/issues/6938.

- The flag `PREFIX` will be deprecated from the `make deps`. It is no longer needed to specify a destination path.

- A flag called `PYTHON_SOURCE` is added to download the python interpreter sources instead of the pre-installed one. This flag can be enabled with values `YES`, `yes`, `y`, `Y`, `1` but is disabled by default.

- The flag `PREFIX` is also deprecated when compiling the Wazuh binaries since the compilation does no longer depend on the installation path. However, it will be also used to call the Makefile for other purposes such as the API installation.

- The flag `USE_FRAMEWORK_LIB` is deprecated. It appears in the RPM SPECS.

- The `libsqlite3.so.0` is no longer installed as well.

Please, ask @wazuh/core for any questions.

Best regards,
Chema.

### Update

- Finally, the `PREFIX` flag is renamed to `INSTALLDIR`. This flag will be needed when the python interpreter has to be built, since the configure script should know where the interpreter will be installed. Here you can see the standard use of this flag:
```
# make deps TARGET=server PYTHON_SOURCE=yes
# make TARGET=server INSTALLDIR=/var/ossec
```

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|#611|

## Description
Closes #611

This  PR modifies the compilation flags as indicated in issue #611, this will allow compiling Wazuh with binaries independent of the installation path.

### RPM Agent
```
15:38:26  Executing(%clean): /bin/sh -e /usr/local/var/tmp/rpm-tmp.BcR142
15:38:26  + umask 022
15:38:26  + cd /build_wazuh/rpmbuild/BUILD
15:38:26  + cd wazuh-agent-4.2.0
15:38:26  + rm -fr /build_wazuh/rpmbuild/BUILDROOT/wazuh-agent-4.2.0-change.installdir.x86_64
15:38:26  + RPM_EC=0
15:38:26  ++ jobs -p
15:38:26  + exit 0
15:38:26  + [[ no == \y\e\s ]]
15:38:26  + [[ no == \y\e\s ]]
15:38:26  + find /build_wazuh/rpmbuild/RPMS/x86_64 -maxdepth 3 -type f -name 'wazuh-agent-4.2.0-change.installdir*' -exec mv '{}' /var/local/wazuh ';'
```

### DEB Agent
```
15:37:27  Finished running lintian.
15:37:27  WARNING generated by debuild:
15:37:27  Making debian/rules executable!
15:37:27  + deb_file=wazuh-agent_4.2.0-change.installdir_amd64.deb
15:37:27  + pkg_path=/build_wazuh/agent
15:37:27  + [[ no == \y\e\s ]]
15:37:27  + mv /build_wazuh/agent/wazuh-agent_4.2.0-change.installdir_amd64.deb /var/local/wazuh
```

### Solaris 10 Agent

```
15:42:41  [0;32m            "    solaris10_cmake: ## Validating control scripts.",[0m
15:42:41  [0;32m            "    solaris10_cmake: ## Packaging complete.",[0m
15:42:41  [0;32m            "    solaris10_cmake: Transferring <",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-agent",[0m
15:42:41  [0;32m            "    solaris10_cmake: > package instance",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-modulesd not running...",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-logcollector not running...",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-syscheckd not running...",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-agentd not running...",[0m
15:42:41  [0;32m            "    solaris10_cmake: wazuh-execd not running...",[0m
15:42:41  [0;32m            "    solaris10_cmake:  v4.2.0 Stopped"[0m
15:42:41  [0;32m        ][0m
15:42:41  [0;32m    }[0m
15:42:41  [0;32m}[0m
```

### Solaris 11 Agent
```
12:33:43  [0;32m            "    solaris11_cmake: Building the package wazuh-agent_v4.2.0-sol11-i386.p5p",[0m
12:33:43  [0;32m            "    solaris11_cmake: Retrieving packages for publisher wazuh ...",[0m
12:33:43  [0;32m            "    solaris11_cmake: Retrieving catalog 'wazuh' ...",[0m
12:33:43  [0;32m            "    solaris11_cmake:  Done",[0m
12:33:43  [0;32m            "    solaris11_cmake: Retrieving and evaluating 1 package(s)...",[0m
12:33:43  [0;32m            "    solaris11_cmake: Fetching manifests: 1/1  100% complete",[0m
12:33:43  [0;32m            "    solaris11_cmake: Fetching manifests: 1/1  100% complete",[0m
12:33:43  [0;32m            "    solaris11_cmake: Download:  0/82 items   0.0/51.9MB  0% complete ",[0m
12:33:43  [0;32m            "    solaris11_cmake: Download: Completed 51.91 MB in 1.72 seconds (30.1M/s)",[0m
12:33:43  [0;32m            "    solaris11_cmake: Archiving:   1/152 items   0.0/50.0MB  0% complete",[0m
12:33:43  [0;32m            "    solaris11_cmake: Archiving: Completed 49.98 MB in 0.25 seconds",[0m
12:33:43  [0;32m            "    solaris11_cmake: /export/home/vagrant/build/solaris11",[0m
12:33:43  [0;32m            "    solaris11_cmake: sudo: PAM account management error: Permission denied",[0m
12:33:43  [0;32m            "    solaris11_cmake: sudo",[0m
12:33:43  [0;32m            "    solaris11_cmake: : ",[0m
12:33:43  [0;32m            "    solaris11_cmake: PAM account management error: Permission denied",[0m
12:33:43  [0;32m            "    solaris11_cmake: sudo",[0m
12:33:43  [0;32m            "    solaris11_cmake: : ",[0m
12:33:43  [0;32m            "    solaris11_cmake: PAM account management error: Permission denied",[0m
12:33:43  [0;32m            "    solaris11_cmake: sudo",[0m
12:33:43  [0;32m            "    solaris11_cmake: : ",[0m
12:33:43  [0;32m            "    solaris11_cmake: PAM account management error: Permission denied",[0m
12:33:43  [0;32m            "    solaris11_cmake: sudo",[0m
12:33:43  [0;32m            "    solaris11_cmake: : ",[0m
12:33:43  [0;32m            "    solaris11_cmake: PAM account management error: Permission denied",[0m
12:33:43  [0;32m            "    solaris11_cmake:  Done",[0m
12:33:43  [0;32m            "    solaris11_cmake: ",[0m
12:33:43  [0;32m            "    solaris11_cmake: pkg unset-publisher: Removal failed for 'wazuh': Unknown publisher 'wazuh'.",[0m
12:33:43  [0;32m            "    solaris11_cmake: umount",[0m
12:33:43  [0;32m            "    solaris11_cmake: : warning: ",[0m
12:33:43  [0;32m            "    solaris11_cmake: /wazuh",[0m
12:33:43  [0;32m            "    solaris11_cmake:  not in mnttab",[0m
12:33:43  [0;32m            "    solaris11_cmake: umount",[0m
12:33:43  [0;32m            "    solaris11_cmake: : ",[0m
12:33:43  [0;32m            "    solaris11_cmake: /wazuh",[0m
12:33:43  [0;32m            "    solaris11_cmake:  not mounted"[0m
```

### macOS Agent
```
12:37:36  [0;33mBuild Successful (3 seconds)[0m
12:37:36  [0;33mThe wazuh agent package for MacOS X has been successfully built.[0m
12:37:36  [0;33mproductsign: using timestamp authority for signature[0m
12:37:36  [0;33mproductsign: signing product with identity "****" from keychain ****[0m
12:37:36  [0;33mproductsign: adding certificate "Developer ID Certification Authority"[0m
12:37:36  [0;33mproductsign: adding certificate "Apple Root CA"[0m
12:37:36  [0;33mproductsign: Wrote signed product archive to /Users/****/packages/warehouse-test/macos/wazuh-agent-4.2.0-change.installdir.pkg.signed[0m
12:37:36  [0;33mThe package /Users/****/packages/warehouse-test/macos/wazuh-agent-4.2.0-change.installdir.pkg was successfully upload for notarization.[0m
12:37:36  [0;33mWaiting 120s to get the results[0m
12:37:36  [0;33mNotarization ticket:[0m
12:37:36  [0;33mNo errors getting notarization info.[0m
12:37:36  [0;33m[0m
12:37:36  [0;33m          Date: 2021-03-03 11:35:33 +0000[0m
12:37:36  [0;33m          Hash: 89962cf9a1c6e092f2be372a307e903cad6f0f5ace57b1e67e4ccb92d18ab296[0m
12:37:36  [0;33m    LogFileURL: ****
12:37:36  [0;33m   RequestUUID: 97c8e9c1-98e8-42ee-a855-d1ac593e6180[0m
12:37:36  [0;33m        Status: success[0m
12:37:36  [0;33m   Status Code: 0[0m
12:37:36  [0;33mStatus Message: Package Approved[0m
12:37:36  [0;33m[0m
12:37:36  [0;33mPackage is notarized and ready to go.[0m
```

## Tests

- Build the package in any supported platform
  - [x] Linux
  - [x] macOS
  - [x] Solaris
- [x] Package installation
- [x] Package upgrade
- [x] Package downgrade
- [x] Package remove
- [x] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] Package install/remove/install
  - [x] Package install/purge/install
  - [x] Check file permissions after installing the package
- Tests for macOS
  - [x] Test the package macOS Sierra
- Tests for Solaris
  - [x] Test the package on Solaris 10
  - [x] Test the package on Solaris 11
  - [x] Check file permissions on Solaris 11 template

### CODE REVIEW

**Review by alberpilot — APPROVED** [review]
LGTM

File: macos/package_files/5.0.0/build.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ function build() {
         make -C ${SOURCES_PATH}/src deps TARGET=agent
 
         echo "Generating Wazuh executables"
-        make -j$JOBS -C ${SOURCES_PATH}/src DYLD_FORCE_FLAT_NAMESPACE=1 TARGET=agent PREFIX=${DESTINATION_PATH} build
+        make -j$JOBS -C ${SOURCES_PATH}/src DYLD_FORCE_FLAT_NAMESPACE=1 TARGET=agent build
```
Comment (okynos) [inline macos/package_files/5.0.0/build.sh:41]: `src/LOCATION`.

File: rpms/SPECS/5.0.0/wazuh-agent-5.0.0.spec [hunk 2]
```diff
@@ -42,14 +42,14 @@ make clean
...
-    make -j%{_threads} TARGET=agent USE_SELINUX=yes PREFIX=%{_localstatedir} DEBUG=%{_debugenabled}
+    make -j%{_threads} TARGET=agent USE_SELINUX=yes DEBUG=%{_debugenabled}
 %else
     %ifnarch x86_64
       MSGPACK="USE_MSGPACK_OPT=no"
     %endif
     deps_version=`cat Makefile | grep "DEPS_VERSION =" | cut -d " " -f 3`
     make deps RESOURCES_URL=http://packages.wazuh.com/deps/${deps_version} TARGET=agent
-    make -j%{_threads} TARGET=agent USE_AUDIT=no USE_SELINUX=yes USE_EXEC_ENVIRON=no PREFIX=%{_localstatedir} DEBUG=%{_debugenabled} ${MSGPACK}
+    make -j%{_threads} TARGET=agent USE_AUDIT=no USE_SELINUX=yes USE_EXEC_ENVIRON=no DEBUG=%{_debugenabled} ${MSGPACK}
```
Comment (okynos) [inline rpms/SPECS/5.0.0/wazuh-agent-5.0.0.spec:52]: `src/LOCATION`.

File: rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec [hunk 3]
```diff
@@ -38,8 +38,8 @@ pushd src
 make clean
 
 # Build Wazuh sources
-make deps PREFIX=%{_localstatedir} TARGET=server
-make -j%{_threads} TARGET=server USE_SELINUX=yes USE_FRAMEWORK_LIB=yes PREFIX=%{_localstatedir} DEBUG=%{_debugenabled}
+make deps TARGET=server
+make -j%{_threads} TARGET=server USE_SELINUX=yes DEBUG=%{_debugenabled}
```
Comment (okynos) [inline rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:42]: Is `USE_FAMEWORK_LIB` also out?

Comment (okynos) [inline rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:42]: `src/LOCATION`.

File: solaris/solaris11/generate_wazuh_packages.sh [hunk 4]
```diff
@@ -133,10 +133,18 @@ compile() {
...
+        else
+            gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no DISABLE_SHARED=yes || return 1
+        fi
     else
-        gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no DISABLE_SHARED=yes || exit 1
+        if [ "$arch" = "sparc" ]; then
+            gmake -j $THREADS TARGET=agent USE_SELINUX=no USE_BIG_ENDIAN=yes DISABLE_SHARED=yes || return 1
+        else
+            gmake -j $THREADS TARGET=agent USE_SELINUX=no DISABLE_SHARED=yes || return 1
+        fi
```
Comment (okynos) [inline solaris/solaris11/generate_wazuh_packages.sh:147]: 4.1 is supported by 4.1 branch.

File: solaris/solaris10/generate_wazuh_packages.sh [hunk 5]
```diff
@@ -146,10 +146,19 @@ installation(){
...
+        else
+            gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no DISABLE_SHARED=yes || return 1
+        fi
     else
-        gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no DISABLE_SHARED=yes || return 1
+        if [ "$arch" = "sparc" ]; then
+            gmake -j $THREADS TARGET=agent USE_SELINUX=no USE_BIG_ENDIAN=yes DISABLE_SHARED=yes || return 1
+        else
+            gmake -j $THREADS TARGET=agent USE_SELINUX=no DISABLE_SHARED=yes || return 1
+        fi
```
Comment (okynos) [inline solaris/solaris10/generate_wazuh_packages.sh:161]: 4.1 is supported by 4.1 branch.

**Review by okynos — CHANGES_REQUESTED** [review]
Some minor changes, GJ.

File: rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec (hunk 3 above)
Comment (DFolchA) [reply rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:42]: As indicated in the issue #611 `USE_FAMEWORK_LIB` is deprecated.

File: debs/SPECS/5.0.0/wazuh-agent/debian/rules [hunk 6]
```diff
@@ -41,7 +41,7 @@ override_dh_install:
 
 	# Build the binaries
 	make -C src deps TARGET=agent
-	make -j$(JOBS) -C src/ TARGET=agent USE_SELINUX=yes PREFIX=$(INSTALLATION_DIR) DEBUG=$(DEBUG_ENABLED)
+	make -j$(JOBS) -C src/ TARGET=agent USE_SELINUX=yes DEBUG=$(DEBUG_ENABLED)
```
[the remark this answers is not in the document]
Comment (DFolchA) [reply debs/SPECS/5.0.0/wazuh-agent/debian/rules:44]: Done in 894f914

File: macos/package_files/5.0.0/build.sh (hunk 1 above)
Comment (DFolchA) [reply macos/package_files/5.0.0/build.sh:41]: Done in 894f914

File: rpms/SPECS/5.0.0/wazuh-agent-5.0.0.spec (hunk 2 above)
Comment (DFolchA) [reply rpms/SPECS/5.0.0/wazuh-agent-5.0.0.spec:52]: Done in 894f914

File: rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec (hunk 3 above)
Comment (DFolchA) [reply rpms/SPECS/5.0.0/wazuh-manager-5.0.0.spec:42]: Done in 894f914

File: solaris/solaris10/generate_wazuh_packages.sh (hunk 5 above)
Comment (DFolchA) [reply solaris/solaris10/generate_wazuh_packages.sh]: Done in 9d7e86d

File: solaris/solaris11/generate_wazuh_packages.sh (hunk 4 above)
Comment (DFolchA) [reply solaris/solaris11/generate_wazuh_packages.sh]: Done in 9d7e86d

File: solaris/solaris11/generate_wazuh_packages.sh [hunk 7]
```diff
@@ -134,9 +134,9 @@ compile() {
     arch="$(uname -p)"
     # Build the binaries
     if [ "$arch" = "sparc" ]; then
-        gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no USE_BIG_ENDIAN=yes DISABLE_SHARED=yes || exit 1
+        gmake -j $THREADS TARGET=agent USE_SELINUX=no USE_BIG_ENDIAN=yes DISABLE_SHARED=yes || return 1
     else
-        gmake -j $THREADS TARGET=agent PREFIX=${install_path} USE_SELINUX=no DISABLE_SHARED=yes || exit 1
+        gmake -j $THREADS TARGET=agent USE_SELINUX=no DISABLE_SHARED=yes || return 1
```
Comment (okynos) [inline solaris/solaris11/generate_wazuh_packages.sh:139]: Test very well this return please.

**Review by okynos — CHANGES_REQUESTED** [review]
Review the testing phase please, the rest is OK, good job.

**Review by VictorMorenoJimenez — APPROVED** [review]
lgtm :rocket:

File: solaris/solaris11/generate_wazuh_packages.sh (hunk 7 above)
Comment (DFolchA) [reply solaris/solaris11/generate_wazuh_packages.sh]: Reverted change in 79f14cb

**Review by okynos — APPROVED** [review]
LGTM.

**Pull request merged**